GitHub workflows failing for conda environment

**Current Behavior**  
As of the most recent merge of a PR into master, [the GitHub workflow that tries to install this repo's conda environment has started failing](https://github.com/nextstrain/conda/runs/535222998#step:4:1). The error message specifically is:

```
Run source $CONDA/etc/profile.d/conda.sh
  source $CONDA/etc/profile.d/conda.sh
  conda env create -f nextstrain.yml
  conda activate nextstrain
  npm install --global auspice
  shell: /bin/bash -e {0}

NotWritableError: The current user does not have write permissions to a required path.
  path: /usr/local/miniconda/envs/.conda_envs_dir_test
  uid: 501
  gid: 20

If you feel that permissions on this path are set incorrectly, you can manually
change them by executing

  $ sudo chown 501:20 /usr/local/miniconda/envs/.conda_envs_dir_test

In general, it's not advisable to use 'sudo conda'.


##[error]Process completed with exit code 1.
```

The same conda environment [passed the workflow tests just the day before on the PR branch](https://github.com/nextstrain/conda/runs/499393690), suggesting that the problem is with the workflow environment.

**Possible solution**  

Some light googling turned up [this exchange on GitHub issues for conda](https://github.com/conda/conda/issues/7267).

Another guess is that GitHub [might have just upgraded their workflow environments to Catalina](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources). But I haven't found evidence of when that change happened.

Since Catalina uses zsh by default instead of bash, it is possible that [the current sourcing of the conda profile script does not activate conda as expected](https://github.com/nextstrain/conda/blob/master/.github/workflows/test-and-publish.yml#L24).

--- a/.github/workflows/test-and-publish.yml
+++ b/.github/workflows/test-and-publish.yml
@@ -16,7 +16,7 @@ jobs:
       - uses: actions/checkout@v1
 
       - name: Fix Conda permissions on macOS
-        run: sudo chown -R $UID $HOME/.conda
+        run: sudo chown -R $UID $CONDA
         if: matrix.os == 'macos-latest'
 
       - name: Install Nextstrain environment

